client/landing/gutenboarding/stores/domain-suggestions/selectors.ts

```diff
@@ -1,15 +1,63 @@
+/**
+ * External dependencies
+ */
+import { select } from '@wordpress/data';
+
 /**
  * Internal dependencies
  */
+import { STORE_KEY } from '.';
 import { DomainSuggestionQuery } from './types';
 import { State } from './reducer';
-import { getSerializedDomainsSuggestionsQuery } from 'state/domains/suggestions/utils';
+import { stringifyDomainQueryObject } from './utils';
 
 export const getState = ( state: State ) => state;
-export const getDomainSuggestions = ( state: State, queryObject: DomainSuggestionQuery ) => {
-	const serializedQuery = getSerializedDomainsSuggestionsQuery( queryObject );
-	if ( serializedQuery ) {
-		return state.domainSuggestions[ serializedQuery ];
-	}
-	return null;
+
+type DomainSuggestionSelectorOptions = Partial< Exclude< DomainSuggestionQuery, 'query' > >;
+export const getDomainSuggestions = (
+	state: State,
+	search: string,
+	options: Partial< Exclude< DomainSuggestionQuery, 'query' > > = {}
+) => {
+	const normalizedQuery = normalizeDomainSuggestionQuery( search, options );
+
+	// We need to go through the `select` store to get the resolver action
+	return select( STORE_KEY ).__internalGetDomainSuggestions( normalizedQuery );
+};
+
+/**
+ * Do not use this selector. It is for internal use.
+ *
+ * @private
+ */
+export const __internalGetDomainSuggestions = (
+	state: State,
+	queryObject: DomainSuggestionQuery
+) => {
+	return state.domainSuggestions[ stringifyDomainQueryObject( queryObject ) ];
 };
+
+/**
+ * Normalize domain query
+ *
+ * It's important to have a consistent, reproduceable representation of a domains query so that the result can be
+ * stored and retrieved.
+ *
+ * @see client/state/domains/suggestions/utils.js
+ * @see client/components/data/query-domains-suggestions/index.jsx
+ */
+function normalizeDomainSuggestionQuery(
+	search: string,
+	q: DomainSuggestionSelectorOptions
+): DomainSuggestionQuery {
+	return Object.assign(
+		{
+			query: search.toLocaleLowerCase(),
+			include_wordpressdotcom: q.include_wordpressdotcom || false, // @FIXME replace `||` with `??`
+			quantity: q.quantity || 5, // @FIXME replace `||` with `??`
+			vendor: 'variation2_front', // see client/lib/domains/suggestions/index.js
```

Comment:
It is extremely useful when we A/B test our suggestions provider. See for example https://github.com/Automattic/wp-calypso/commit/a0cd2b23950dd9f25912aab0e7a78dbeb493d332#diff-a594ab81317e1efd2a3317ed1bbd1eed Ideally, we only have one, central place where the vendor is selected - at times there are A/B tests, but most of the time it's hardcoded (like right now).

Reply:
That makes sense. It sounds like for a shared package, we'd default to the control (`variation2_front`) and allow it to be overridden by consumers of the package (for an abtest, for example).

I'll adjust this accordingly unless you have more to add.

Reply:
There may also be times where we might want to use a different suggestion vendor based on additional user criteria such as locale. ...just mentioning it so we don't preclude this.

Reply:
> There may also be times where we might want to use a different suggestion vendor based on additional user criteria such as locale. ...just mentioning it so we don't preclude this.

👍 I don't think we'd preclude it, the consumer would handle it.

On the consumer's end, it would look like:

```js
function MyComponent( { domainSearch } ) {

  const suggestions = useSelect( select =>
    select('automattic/domains/suggestions').getDomainSuggestions(
      domainSearch,
      { vendor: getUserLocale() === 'en_GB' ? 'gb-vendor' : 'rest-vendor' }
    )
  );
  // …
}
```

This store (and associated API calls) have no knowledge of locale, does that seem like an oversight?